src/transport/smtp/mod.rs

```diff
@@ -154,6 +154,8 @@ mod async_transport;
 pub mod authentication;
 pub mod client;
 pub mod commands;
+#[cfg(any(feature = "native-tls", feature = "rustls-tls", feature = "boring-tls"))]
```

Comment:
I'd remove this feature gate and instead give runtime errors when parsing the URL if no TLS backend is available. We might also want to document that.

Reply:
I've added runtime checks. I propose to add more elaborate documentation to the generic "from connection URL" method, that also supports sendmail and file transports. I can work on that onze this is merged.